### ISSUE TITLE
DNS-SD: PTR query response does not contain additional records as per RFC 

### ISSUE THREAD

**Issue by Jonas275** [systemd/systemd#24859]
### systemd version the issue has been seen with

251

### Used distribution

Yocto 4.0.2, FSLC Wayland

### Linux kernel version used

5.15.48-5.15.5-1.0.0+gd4e1daa8ee99 (Linux-fslc-imx )

### CPU architectures issue was seen on

aarch64

### Component

systemd-resolved

### Expected behaviour you didn't see

Expecting the DNS-SD response to a PTR query to include already the SRV and TXT records as additional records,
as described in chapter 12 of RFC 6763

https://www.ietf.org/rfc/rfc6763.txt
12.1.  PTR Records

   When including a DNS-SD Service Instance Enumeration or Selective
   Instance Enumeration (subtype) PTR record in a response packet, the
   server/responder SHOULD include the following additional records:

   o  The SRV record(s) named in the PTR rdata.
   o  The TXT record(s) named in the PTR rdata.
   o  All address records (type "A" and "AAAA") named in the SRV rdata.

### Unexpected behaviour you saw

The response to a PTR query does not contain additional records.

**Comment by pemensik** [comment]
I doubt even any of unbound, bind9 or dnsmasq implements such feature. Do you know any authoritative server handling DNS-SD, which would reply to ``dig -t PTR <anyname>`` with SRV and TXT records as well?

Is this request to add ti to domain port stub listener or just mdns subsystem?

**Comment by jonas-priv** [comment]
Hi @pemensik , thanks for the reply!.
It seems avahi implements such feature. We've been using avahi in our embedded products and want to switch to systemd-resolved. Our client app still expects the additional records to discover our devices.

**Comment by pemensik** [comment]
Yes, but avahi is mdns only. If such behaviour is enabled only for DNS-SD names on mDNS protocol and only for [DNS-SD related names](https://www.ietf.org/rfc/rfc6763.html#section-13.1), why not. But extra queries should not be done for common hostname reverse DNS query.

**Comment by jonas-priv** [comment]
Fair enough, if you could add it to the mdns subsystem only, that would solve our problem.

**Comment by pemensik** [comment]
I am not sure limiting this to mdns only is the best thing. I think at least Apple has some support also for DNS-SD in DNS itself, at least dns-sd.org mentions such RFC. I think systemd-resolved should just add SRV and TXT records to additional section of response, if they were already in cache and queried name starts with `_` for example. I think `_` should not be part of hostname, especially not on the label start. That should be clear indication this is not reverse hostname query, but likely related to DNS-SD.
